### PR TITLE
feat(helm): allow setting resources for spire agent and server workloads

### PR DESCRIPTION
**Pull request by sjoukedv**
Added `resources`  to the helm values for `authentication.mutual.spire.install.agent` and `authentication.mutual.spire.install.server`.

### CODE REVIEW

File: USERS.md [hunk 1]
```diff
@@ -665,6 +665,12 @@ Users (Alphabetically)
       L: https://schubergphilis.com/en
       Q: @stimmerman @shoekstra @mbaumann
 
+    * N: SDV Services
+      D: SDV Services uses Cilium to host Wordpress multi-tenant the cloud-native way and also as the CNI for customer Kubernetes clusters.
+      U: CNI, Networking, NetworkPolicy, Hubble, IPAM, Kubernetes services
+      L: https://sdvservices.nl
+      Q: @Sjouke de Vries
+
```
Comment (sayboras) [inline USERS.md]: Thanks for adding this, can you split this change into another PR for faster review and merge?

Comment (sjoukedv) [reply USERS.md]: Done, see https://github.com/cilium/cilium/pull/34746

**Review by tommyp1ckles — APPROVED** [review]

File: install/kubernetes/cilium/values.schema.json [hunk 2]
```diff
@@ -5324,4 +5330,4 @@
   },
   "$schema": "http://json-schema.org/draft-07/schema#",
   "type": "object"
-}
\ No newline at end of file
+}
```
Comment (gandro) [inline install/kubernetes/cilium/values.schema.json]: This newline change seems off. If you use the Makefile to generate the schema (`make -C install/kubernetes`), I believe this should be fixed

**Review by gandro — COMMENTED** [review]
Thanks for the PR!

This looks good overall, but I believe you need to regenerate the Helm documentation too:

```
make -C Documentation update-helm-values
make -C install/kubernetes
```

After applying those changes, make sure to squash them into the existing commit. Thanks!

File: install/kubernetes/cilium/values.schema.json (hunk 2 above)
Comment (sjoukedv) [reply install/kubernetes/cilium/values.schema.json]: My bad that's probably my editor. I also had to change `values.yaml.tmpl` before running the Makefile(s).

**Comment by sjoukedv** [comment]
Replaced by https://github.com/cilium/cilium/pull/34798 due to signed off message missing from commits
